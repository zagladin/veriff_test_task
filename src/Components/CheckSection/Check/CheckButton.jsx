import React from 'react';
import className from 'classnames/bind';
import PropTypes from 'prop-types';
import styles from './CheckButton.module.scss';

const CheckButton = (
  {
    text,
    active,
    onClick,
    disabled,
  },
) => {
  const cx = className.bind(styles);

  const buttonStyles = cx({
    checkButton: true,
    inactive: !active,
  });

  return (
    <button
      className={buttonStyles}
      onClick={onClick}
      disabled={disabled}
    >
      {text}
    </button>
  );
};

CheckButton.propTypes = {
  disabled: PropTypes.bool.isRequired,
  active: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default React.memo(CheckButton);
