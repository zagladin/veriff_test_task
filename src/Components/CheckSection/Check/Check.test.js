import React from 'react';
import renderer from 'react-test-renderer';
import CheckButton from './CheckButton';

const mockData = {
  onClick: () => {
  },
  text: 'Test for button',
  disabled: false,
  active: true,
};

it('renders correctly with one class', () => {
  const checkButton = renderer
      .create(
          <CheckButton
              onClick={mockData.onClick}
              text={mockData.text}
              disabled={mockData.disabled}
              active={mockData.active}/>,
      )
      .toJSON();
  expect(checkButton).toMatchSnapshot();
});

it('renders correctly with two classes', () => {
  const data = {...mockData, active: false};
  const checkButton = renderer
      .create(
          <CheckButton
              onClick={data.onClick}
              text={data.text}
              disabled={data.disabled}
              active={data.active}/>,
      )
      .toJSON();
  expect(checkButton).toMatchSnapshot();
});
