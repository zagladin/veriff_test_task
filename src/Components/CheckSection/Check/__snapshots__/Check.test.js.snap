// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with one class 1`] = `
<button
  className="checkButton"
  disabled={false}
  onClick={[Function]}
>
  Test for button
</button>
`;

exports[`renders correctly with two classes 1`] = `
<button
  className="checkButton inactive"
  disabled={false}
  onClick={[Function]}
>
  Test for button
</button>
`;
